src/transformers/modeling_utils.py

```diff
@@ -2062,12 +2127,18 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
 
             logger.info("Detected DeepSpeed ZeRO-3: activating zero.init() for this model")
             init_contexts = [deepspeed.zero.Init(config_dict_or_path=deepspeed_config())] + init_contexts
+        elif load_in_8bit:
+            init_contexts = [init_empty_weights()]  # Force enable init empty weights
+            logger.info("Detected 8-bit loading: activating 8-bit loading for this model")
         elif low_cpu_mem_usage:
             init_contexts.append(init_empty_weights())
 
         with ContextManagers(init_contexts):
             model = cls(config, *model_args, **model_kwargs)
 
+        if load_in_8bit:
+            model = replace_8bit_linear(model, threshold=int8_threshold)
```

Comment:
Or the logger.info could go here actually!

Reply:
Yes! Thanks for the proposal!
Resolving this because of: a92564137ec21e665c8e241251d5a2c565113b20